mlflow/server/handlers.py

```diff
@@ -3784,6 +3796,209 @@ def _get_dataset_records_handler(dataset_id):
     return _wrap_response(response_message)
 
 
+# Prompt Optimization Job Management
+class PromptOptimizationJobManager:
+    """Manages prompt optimization jobs using a thread pool for controlled concurrency."""
+
+    def __init__(self):
+        self._jobs: Dict[str, Dict] = {}
+        self._next_job_id = 0
+        self._lock = threading.Lock()
+        self._executor = ThreadPoolExecutor(
+            max_workers=MLFLOW_MAX_CONCURRENT_PROMPT_OPTIMIZATION_JOBS.get(),
+            thread_name_prefix="prompt_opt"
+        )
+
+    def create_job(self, train_dataset_id: str, eval_dataset_id: str,
+                   prompt_url: str, scorers: list[str], target_llm: str = None, 
+                   algorithm: str = None) -> str:
```

Comment:
```suggestion
                   algorithm: str | None = None) -> str:
```

Reply:
Same for `target_llm`

Reply:
`target_llm` should be required (no default value)